ddtrace/internal/_encoding.pyx

```diff
@@ -580,10 +581,11 @@ cdef class MsgpackEncoderV03(MsgpackEncoderBase):
             ret = pack_number(&self.pk, span.trace_id)
             if ret != 0: return ret
 
-            ret = pack_bytes(&self.pk, <char *> b"parent_id", 9)
-            if ret != 0: return ret
-            ret = pack_number(&self.pk, span.parent_id)
-            if ret != 0: return ret
+            if span.parent_id is not None:
```

Comment:
`if has_parent_id:` is likely the faster comparison to do here, vs checking the PyObject type.

Reply:
actually, I am not 100% sure how Cython compiles them, but my best guess is that checking the `bint` is better/faster.

Reply:
hmm I'll give it a try